### DISCUSSION TITLE
Proper way to send 1 file from main to all open branches

### DISCUSSION THREAD

**Discussion by Hifilo** [community/community#59746] [New to GitHub]
Hello. 

This is a general workflow question.

Say I have multiple branches open with files in various states, and say some of those branches also have branches.

I then decide to update my readme file. How should I go about getting the readme file from vscode where I've written it to all the other open branches without creating any conflicts?

In my situation, I went and wrote up the readme in vscode, and then I synced up from local main to origin main.

I was able to rebase the other branches that are in line with origin main easily.

I have 1 branch that is not in sync with main. It is older code, that I may want to return to, or I will delete, but I'm not 100% sure yet. However I want to get this 1 readme file to that branch since it isn't related to the code files

1. How can I get this 1 file from the main branch to the other branch without conflict, _edit: and without updating any of the other files?_

2. Is there a way to update that entire branch **_without_** updating the files with the old code?

Apologies in advance if this question contains an XY problem. 😬

Thanks.

**Comment by GhostOf0days** [comment]


**Reply by marnen** [reply]
Better yet, put the README change into a commit on a branch of its own, and merge it into whatever branches you like. 

Don’t copy and paste when you could sling commits.

**Reply by Hifilo** [reply]
Ah I see how to work with it now.

This actually clears up a ton of workflow questions I wasn't entirely clear on. 

Much appreciated.

**Reply by Hifilo** [reply]
actually @marnen perhaps I'm missing something here. This didn't work the way I was expecting it to.

When I create a branch with `git branch readme`, it makes a branch that is up to date. So when I update the readme, merge it into the older branch, it merges all the files, updating the files I don't want to update. Can you elaborate on the process?

**Reply by marnen** [reply]
Did you commit the README file changes to the readme branch?

**Reply by Hifilo** [reply]
yes I updated the file, committed, & pushed.

I then went to my older branch to merge the readme branch. when I merged it updated the readme file, & everything else.

**Reply by marnen** [reply]
Also, remember that branches go back to the beginning of time. If you want to keep newer commits out of the older branch, either start the README branch from the older state or cherry-pick instead of merging (if you really must). 

At this point I’m going to suggest that you find a good tutorial of how git merge works, or play around with a toy repo.

**Reply by Hifilo** [reply]
> I have 1 branch that is not in sync with main. It is older code, that I may want to return to, or I will delete, but I'm not 100% sure yet. However I want to get this 1 readme file to that branch since it isn't related to the code files
> 
> 1.    How can I get this 1 file from the main branch to the other branch without conflict, edit: and without updating any of the other files?
> 
> 2.    Is there a way to update that entire branch without updating the files with the old code?

`git cherry pick` would allow me to grab that one file?

I guess one alternative would be to create a branch from the older branch, update the readme, then merge. This way there is nothing new except for the readme. Not sure if this saves any time. I'll mess around and see what works for me.

**Reply by marnen** [reply]
The latter is what I would probably do. Cherry-picking will work but will make the commit history less useful.
